### ISSUE TITLE
Joining and sorting ranks fails to load serialized double array holding page rank scores

### ISSUE THREAD

**Issue by sebastian-nagel** [commoncrawl/cc-webgraph#5]
- seen during construction of the [web graphs October, November/December 2021 and January 2022](https://commoncrawl.org/2022/03/host-and-domain-level-web-graphs-oct-nov-jan-2021-2022/)
```
[main] INFO org.commoncrawl.webgraph.JoinSortRanks - Loading page rank values from host/cc-main-2021-22-oct-nov-jan-host-pagerank.ranks
Exception in thread "main" java.lang.IllegalArgumentException: newLimit < 0: (-1216172000 < 0)
        at java.base/java.nio.Buffer.createLimitException(Buffer.java:372)
        at java.base/java.nio.Buffer.limit(Buffer.java:346)
        at java.base/java.nio.ByteBuffer.limit(ByteBuffer.java:1107)
        at java.base/java.nio.MappedByteBuffer.limit(MappedByteBuffer.java:235)
        at java.base/java.nio.MappedByteBuffer.limit(MappedByteBuffer.java:67)
        at it.unimi.dsi.fastutil.io.BinIO.loadDoubles(BinIO.java:6431)
        at it.unimi.dsi.fastutil.io.BinIO.loadDoubles(BinIO.java:6452)
        at it.unimi.dsi.fastutil.io.BinIO.loadDoubles(BinIO.java:6520)
        at it.unimi.dsi.fastutil.io.BinIO.loadDoubles(BinIO.java:7006)
        at it.unimi.dsi.fastutil.io.BinIO.loadDoubles(BinIO.java:7018)
        at org.commoncrawl.webgraph.JoinSortRanks.loadPageRank(JoinSortRanks.java:50)
        at org.commoncrawl.webgraph.JoinSortRanks.main(JoinSortRanks.java:319)
```
- worked around (no time at this point for a deeper analysis) by downgrading (reverting recent commits to f33704b used to build the preceding graph)

**Comment by sebastian-nagel** [comment]
The failure loading of a double array is reproducible with any array if the size of the serialization file is 2 GiB or more, see 49174af. Downgrading fastutil-core to 8.5.7 fixes the issue, the unit test also passes using the current head of fastutil (vigna/fastutil@b813824).

**Comment by sebastian-nagel** [comment]
Fixed by downgrading to fastutil-core 8.5.7

**Comment by vigna** [comment]
Thank you for reporting this. Fixed in fastutil 8.5.9.
